File: 1-js-basics/5.1-while-loop-lab/08-graduation.js
function solve(arr) {

    let name = arr.shift();

    let index = 0;

    let current = arr[index];

    let counter = 0;

    let grade = 1;

    let sum = 0;

    while (counter <= 1) {

        current = Number(current);

        if (grade > 12) {

            break;

        }

        if (current >= 4) {

            sum += current;

            grade++;

        } else {

            counter++;

        }

        index++;

        current = arr[index];

    }

    if (counter > 1) {

        return `${name} has been excluded at ${grade} grade`;

    }

    return `${name} graduated. Average grade: ${(sum / 12).toFixed(2)}`;

}

console.log(solve(["Gosho", 
"5",
"5.5",
"6",
"5.43",
"5.5",
"6",
"5.55",
"5",
"6",
"6",
"5.43",
"5"])
);

File: 2-python/1-python-basics/5.1-while-loop-lab/08-graduation.py
student_name = input()
yearly_grade = float(input())
yearly_grades_sum = 0
grade_counter = 1
failure_counter = 0

while True:
    if yearly_grade >= 4:
        yearly_grades_sum += yearly_grade
        grade_counter += 1
        if grade_counter > 12:
            break
    else:
        failure_counter += 1
        if failure_counter > 1:
            break
    yearly_grade = float(input())

output = ''

if failure_counter > 1:
    output = f'{student_name} has been excluded at {grade_counter} grade'
else:
    avg_grade = yearly_grades_sum / 12
    output = f'{student_name} graduated. Average grade: {avg_grade:.2f}'

print(output)
